### PR TITLE
Diff against tc-proxy

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: proxy.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	return rp.tmpToken
+}
+
+func (rp RelengapiProxy) runForever() {
+	log.Println("Proxying to RelengAPI with permissions:",
+		rp.permissions, "on port", rp.listenPort)
+
+	// httputil's ReverseProxy is not specifically "reverse", and it will
+	// do fine here.  The director transforms outgoing requests.
+	director := func(req *http.Request) {
```
Comment (lightsofapollo-taskcluster) [inline proxy.go:54]: very nice

File: relengapi.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	Metadata    interface{} `json:"metadata,omitempty"`
+	Disabled    bool        `json:"disabled,omitempty"`
+	Permissions []string    `json:"permissions,omitempty"`
+	Description string      `json:"description,omitempty"`
+	User        string      `json:"user,omitempty"`
+	Token       string      `json:"token,omitempty"`
+}
+
+func getTmpToken(url string, issuingToken string, expires time.Time, perms []string) string {
+	// TODO: retry this operation
```
Comment (lightsofapollo-taskcluster) [inline relengapi.go]: pmoore has some retry code

File: relengapi.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		Metadata:    map[string]interface{}{},
+	}
+
+	reqbody, err := json.Marshal(request)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	client := &http.Client{}
+	reqUrl := fmt.Sprintf("%s/tokenauth/tokens", url)
```
Comment (lightsofapollo-taskcluster) [inline relengapi.go]: This is probably fine but you may get better url joins using net/url

File: relengapi.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		log.Fatal(err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if resp.StatusCode != 200 {
+		log.Fatalf("Got '%s' while trying to get new tmp token:\n%s",
```
Comment (lightsofapollo-taskcluster) [inline relengapi.go]: IMO I would return errors rather then logging/fatal'ing them here

File: relengapi.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+			resp.Status, string(body))
+	}
+
+	var responseBody interface{}
+	err = json.Unmarshal(body, &responseBody)
+	if err != nil {
+		log.Fatal(err)
```
Comment (lightsofapollo-taskcluster) [inline relengapi.go]: ^

File: taskcluster.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+func getTaskScopes(taskId string) []string {
+	// We do not need auth for this operation
+	q := queue.New("", "")
+	q.Authenticate = false
+
+	task, callSummary := q.Task(taskId)
+	if callSummary.Error != nil {
+		log.Fatalf("Could not fetch taskcluster task '%s' : %s",
```
Comment (lightsofapollo-taskcluster) [inline taskcluster.go]: nit: IMO I would return this error and let the main control flow deal with fatal'ing out

**Comment by petemoore** [comment]
Should we close this PR?

**Pull request closed without merging**